lib/crawler_web/controllers/keyword_controller.ex

```diff
@@ -0,0 +1,72 @@
+defmodule CrawlerWeb.KeywordController do
+  use CrawlerWeb, :controller
+
+  alias Crawler.Keyword.Helpers.CSVParser
+  alias Crawler.Keyword.Keywords
+  alias Crawler.Keyword.Schemas.KeywordCSVFile
+
+  def create(conn, %{"keyword_csv_file" => params}) do
+    changeset = %{
+      KeywordCSVFile.create_changeset(%KeywordCSVFile{}, params)
+      | action: :validate
+    }
+
+    case changeset.valid? do
+      true ->
+        parse_uploaded_file(conn, changeset.changes)
+
+      false ->
+        show_error_flash_message_and_redirects_to_dasboard(
+          conn,
+          gettext("The file doesn't look like CSV")
+        )
+    end
+  end
+
+  defp parse_uploaded_file(conn, changes) do
+    case CSVParser.parse(changes.file.path) do
+      {:ok, keyword_list} ->
+        Keywords.create_keyword_list(keyword_list, conn.assigns.current_user.id)
```

Comment:
Is there a way to know how many records were created, not sure the `insert_all` return the `record number there` 

Or  at least we should pattern matching here to ensure the action was success

```suggestion
        {num_keyword, xxxx} = Keywords.create_keyword_list(keyword_list, conn.assigns.current_user.id)
```

Reply:
Updated in [1d816b2](https://github.com/vnntsu/elixir-ic/pull/36/commits/1d816b234f96b49e84b553a585e8119e4990ed31)